### DISCUSSION TITLE
Authentication and authorization are a must if Solara can be taken seriously

### DISCUSSION THREAD

**Discussion by metaperl** [widgetti/solara#89] [Q&A]
As [stated elsewhere](https://www.reddit.com/r/Python/comments/13fegbp/comment/jjv344b/?utm_source=share&utm_medium=web2x&context=3) no one doing serious work can leave A and A to chance. I looked at [the routing docs](https://solara.dev/docs/understanding/routing) but did not see how to control access to routes based on authenticated role.

A world class web app framework must be able to render the UI differently based on who is logged in - certain resources wont even be listed. and even if they hard-link to it, access will be denied based on role.

What is the vision for the solara devs in terms of A and A?

**Comment by maartenbreddels** [comment]
As I replied on Redit, we have support for authentication, and since it's fastapi there are many possibilities to roll out yourself. 

Authorization is difficult to have a one-size-fits-all solution for, however without having an example, it may be difficult to imagine how to do that with solara. I've opened https://github.com/widgetti/solara/pull/91 (which is now live at https://solara.dev/apps/authorization  )
Let us know if that is what you had in mind. Maybe for completeness, it should not list the admin page/tab if not logged in and not a user?

Hope that answers your question.

**Comment by maartenbreddels** [comment]
Note that currently, the link to the source code is not valid because the branch is not merged. The source code can now be found at https://github.com/widgetti/solara/tree/docs_example_authorization/solara/website/pages/apps/authorization
